app/agents/harness/store/workspace_store.py

```
@@ -1043,6 +1122,168 @@ async def list_files(self, path: str = "/", depth: int = 2) -> List[Dict[str, An
+
+        parsed.sort(key=lambda pair: pair[1], reverse=True)
+        to_delete = parsed[keep:]
+        if not to_delete:
+            return
+
+        for session_id, _ in to_delete:
+            await self._delete_session_workspace(session_id=session_id)
+            try:
+                await self.delete_file(f"/user/sessions/{session_id}.json")
+            except Exception:
+                pass
```

Ruff (0.14.13)

`try`-`except`-`pass` detected, consider logging the exception

(S110)